src/caffe/util/io.cpp

```diff
@@ -72,34 +72,21 @@ void WriteProtoToBinaryFile(const Message& proto, const char* filename) {
 #ifdef USE_OPENCV
 cv::Mat ReadImageToCVMat(const string& filename,
     const int height, const int width, const bool is_color) {
+  cv::Mat orig_cv_img = ReadImageToCVMat(filename, is_color);
+  if (height == 0 || width == 0) { return orig_cv_img; }
   cv::Mat cv_img;
-  int cv_read_flag = (is_color ? CV_LOAD_IMAGE_COLOR :
-    CV_LOAD_IMAGE_GRAYSCALE);
-  cv::Mat cv_img_origin = cv::imread(filename, cv_read_flag);
-  if (!cv_img_origin.data) {
-    LOG(ERROR) << "Could not open or find file " << filename;
-    return cv_img_origin;
-  }
-  if (height > 0 && width > 0) {
-    cv::resize(cv_img_origin, cv_img, cv::Size(width, height));
-  } else {
-    cv_img = cv_img_origin;
```

Comment:
Aside from style cleanup, this PR also eliminates this unnecessary copy in the no resize case by returning immediately on line 76.  (Not sure whether it's significant, but the performance should be strictly better.)